src/base/utils/fs.h

```diff
@@ -33,12 +33,40 @@
  * Utility functions related to file system.
  */
 
+#include <iterator>
+#include <memory>
+
 #include <QString>
 
+class QByteArray;
+class QFileDevice;
+
 namespace Utils
 {
     namespace Fs
     {
+        // A wrapper class that satisfy LegacyOutputIterator requirement
+        class FileDeviceOutputIterator
+            : public std::iterator<std::output_iterator_tag, void, void, void, void>
+        {
+        public:
+            explicit FileDeviceOutputIterator(QFileDevice &device, const int bufferSize = (4 * 1024));
```

Comment:
I wouldn't still limit it to file devices only. It can have broader use without any modifications... Just rename it to IODeviceOutputIterator.

Reply:
Also this class is IO related (not filesystem) so it isn't correct place for it.

Reply:
>I wouldn't still limit it to file devices only. It can have broader use without any modifications

QIODevice is also inherited by network socket and I'm not sure this class is suitable for them, so I would like to limit its usage. If the need arises, then it can be changed later.

>Also this class is IO related (not filesystem) so it isn't correct place for it.

Please advise where to place it.

Reply:
>I would like to limit its usage. If the need arises, then it can be changed later.

Ok.
>Please advise where to place it.

In filedeviceoutputiterator.h/.cpp directly in src/base?
Or you prefer to create namespace for it? We can do it later if there are similar IO classes appear.

Reply:
>Or you prefer to create namespace for it? We can do it later if there are similar IO classes appear.

I'm interested in creating a namespace, what about at src/base/utils/io.h? And it will be `Utils::IO::FileDeviceOutputIterator`.

Reply:
Ok.

Reply:
Updated.